test/integration/trim_mode_test.rb

```diff
@@ -528,4 +528,32 @@ def test_raw_output
     END_EXPECTED
     assert_template_result(expected, text)
   end
+
+  def test_pre_trim_blank_preceding_text
```

Comment:
pushed a commit to port this test too, which is the corresponding test without the `bug_compatible_whitespace_trimming: true` option